channels/ON-CHAIN.md

```diff
@@ -542,3 +546,172 @@ Since the miner is expending resources for the contract's execution, the gas
+`lock_period` ahead. That is, by `chain_height == 1011` the `lock_period` is set
+to run out at `chain_height == 1111`.
+
+Now it is important to note, that at even at `chain_height := 1110` Alice can
+still dispute the *initial* update issued by Bob at `chain_height := 1000` by
+providing a co-signed payload with `round := 24` or higher but her dispute would
+bump the lock by `lock_period` too.
+
+
+### `channel_slash`
+
+If a malicious party sent a `channel_close_solo` or `channel_force_progress_tx`
+with an outdated state, the honest party has the opportunity to issue a
+`channel_slash` transaction. This transaction MUST include a state with a higher
+`round` number than the one being disputed, signed by all peers for a successful
+challenge.
+
+Serialization defined [here](../serializations.md#channel-slash-transaction)
+
+
+- `channel_id`: channel id as recorded on-chain
+- `from_id`: channel participant or delegate that posts the slashing transaction
+- `payload`: transaction proving that the proof of inclusion is part of the
+  channel
+- `poi`: proof of inclusion
+- `ttl`: blockheight target until which this transaction can be included
+- `fee`: transaction fee
+- `nonce`: taken from the `from_id`'s account
+
+The proof of inclusion represents the channel's internal state. It has to
+include both participants' accounts and their balances.
+If there are any contracts in the channel and those have balances of their own,
+they are not provided in the proof of inclusion but they are rather to be force
+pushed in subsequent transactions. It is up to participants to decide if they
+want to post them at all. Thus the accumulative balances of the accounts in the
+slash transaction can be lower than the channel balance persisted on-chain.
+
+Payload is a valid transaction that has:
+
+* `state_hash` equal to the proof of inclusion's root hash
+* `channel_id` being the same as the transaction `channel_id`
+* `round` greater than the last on-chain `round` for that channel id. If the
+  state tree entry has a `solo_round` bigger than the `round` state tree, then
+  this slash will invalidate progress produced on-chain
+
+The payload can be either empty or a signed transaction.
+
+If the payload is empty, the channel is closed according to the last on-chain
+state. In this case the proof of inclusion root must be equal to the one
+persisted for the channel on-chain.
+
+If the payload is a transaction it MUST be a `channel_offchain_tx`. It MUST be
+co-signed.
+
+
+#### Requirements
+
+MUST be signed using private key corresponding to the public key `from_id`.
+
+
+## Channel state tree
+
+Each block MUST commit to a Merkle Patricia tree of open channels, where the
+`channel_id` specifies the path.
+At a leaf, nodes store information pertaining to the current state of the given
+channel.
+
+- `channel_id`
+- `initiator_pubkey`
+- `responder_pubkey`
+- `total_amount`
+- `initiator_amount`
```

Comment:
I see that [serialization doc](https://github.com/aeternity/protocol/blob/2d3246e9bed457f4399e4dde1b3e29ee3af11914/serializations.md#channel) (linked below) has `responder_amount` while absent here. Any reasons?

Reply:
I see added in dc3fa1b. Thanks.